tests/vcs/mercurial/HgCommitBearTest.py

```diff
@@ -0,0 +1,504 @@
+from coala_utils.string_processing.Core import escape
+from coalib.misc.Shell import run_shell_command
+from coalib.settings.ConfigurationGathering import get_config_directory
+from coalib.settings.Section import Section
+from coalib.settings.Setting import Setting
+
+
+@generate_skip_decorator(HgCommitBear)
+class HgCommitBearTest(unittest.TestCase):
+
+    testFile = 'testFile.txt'
+
+    @staticmethod
+    def run_hg_command(*args, stdin=None):
+        run_shell_command(' '.join(('hg',) + args), stdin)
+
+    @staticmethod
+    def hg_commit(msg):
+        if(msg == ''):
+            msg = ' '
+        with open(HgCommitBearTest.testFile, 'w') as f:
+            f.write(msg)
+        HgCommitBearTest.run_hg_command('commit -l ',
+                                        HgCommitBearTest.testFile)
+
+    def changeHgConfig(self, newConfig):
+        with open(self.hgdir+'/.hg/hgrc', 'w') as f:
+            f.write(newConfig)
+
+    def run_uut(self, *args, **kwargs):
+        """
+        Runs the unit-under-test (via `self.uut.run()`) and collects the
+        messages of the yielded results as a list.
+
+        :param args:   Positional arguments to forward to the run function.
+        :param kwargs: Keyword arguments to forward to the run function.
+        :return:       A list of the message strings.
+        """
+        return list(result.message for result in self.uut.run(*args, **kwargs))
+
+    def assert_no_msgs(self):
+        """
+        Assert that there are no messages in the message queue of the bear, and
+        show the messages in the failure message if it is not empty.
+        """
+        self.assertTrue(
+            self.msg_queue.empty(),
+            'Expected no messages in bear message queue, but got: ' +
+            str(list(str(i) for i in self.msg_queue.queue)))
+
+    def setUp(self):
+        self.msg_queue = Queue()
+        self.section = Section('')
+        self.uut = HgCommitBear(None, self.section, self.msg_queue)
+
+        self._old_cwd = os.getcwd()
+        self.hgdir = mkdtemp()
+        os.chdir(self.hgdir)
+        self.run_hg_command('init')
+        run_shell_command('touch '+HgCommitBearTest.testFile)
+        self.run_hg_command('add', HgCommitBearTest.testFile)
+        newConfigFile = ('[ui]\n'
+                         'username = user.email <coala@coala.io>')
+        self.changeHgConfig(newConfigFile)
+        self.run_hg_command('commit -m "testInit Commit"')
+
+    @staticmethod
+    def _windows_rmtree_remove_readonly(func, path, excinfo):
+        os.chmod(path, stat.S_IWRITE)
+        func(path)
+
+    def tearDown(self):
+        os.chdir(self._old_cwd)
+        if platform.system() == 'Windows':
+            onerror = self._windows_rmtree_remove_readonly
+        else:
+            onerror = None
+        shutil.rmtree(self.hgdir, onerror=onerror)
+
+    def test_check_prerequisites(self):
+        _shutil_which = shutil.which
```

Comment:
It would be better if this is in a new `CommitBearTest` , as the modification of `shutil.which` is a mock anyway, if not better as it is less confusing.

Move the git test as-is (no modification) to the new `CommitBearTest`.
After merge we need to create an issue for someone to rewrite it to use proper mocking.

Reply:
So should I let the Hg one remain here or will a test be also kept for this one later ?

Reply:
Since coverage won't be complete without this one for both git and hg,
I am letting the hg check remain here and moving the git check in the super-class

Reply:
> After merge we need to create an issue for someone to rewrite it to use proper mocking.

Raised https://github.com/coala/coala-bears/issues/2476